ksql-metastore/src/main/java/io/confluent/ksql/metastore/MetaStoreImpl.java

```diff
@@ -83,8 +83,18 @@ public StructuredDataSource getSource(final String sourceName) {
   public Optional<StructuredDataSource> getSourceForTopic(final String ksqlTopicName) {
     return dataSources.values()
         .stream()
-        .filter(p -> p.source.getKsqlTopic().getName() != null
-            && p.source.getKsqlTopic().getName().equals(ksqlTopicName))
+        .filter(p -> p.source.getTopicName() != null
+            && p.source.getTopicName().equals(ksqlTopicName))
+        .map(sourceInfo -> sourceInfo.source)
+        .findFirst();
+  }
+
+  @Override
+  public Optional<StructuredDataSource> getSourceForKafkaTopic(final String kafkaTopicName) {
```

Comment:
Same as the comment above, this should return a collection.

Reply:
Done.